R/mod_gene_var.R

```diff
@@ -101,16 +110,46 @@ mod_gene_variant_server <- function(input, output, session, specimens){
       pointSize = 2.5
     )
     
-    # maftools::oncoplot(kairos::jhu_tumor_file, 
-    #                    genes = c("NF1", "EGR2"),
-    #                    drawRowBar = TRUE, drawColBar = TRUE,
-    #                    clinicalFeatures = c("tumorType", "sampleType","individualID"),
-    #                    #sortByAnnotation = TRUE, annotationColor = color_list, groupAnnotationBySize = FALSE, 
-    #                    removeNonMutated = FALSE, logColBar = FALSE,
-    #                    SampleNamefont = 5,annotationFontSize = 1.5, fontSize = 1,legendFontSize = 1.8,
-    #                    titleFontSize = 1.0, keepGeneOrder = TRUE, GeneOrderSort = TRUE, bgCol = "white", borderCol = "white")
+    
   })
   
+
+  output$onco_plot <- shiny::renderPlot({
+  
+    tumor_sample_bc <- as.vector(kairos::jhu_tumor_file@data$Tumor_Sample_Barcode[kairos::jhu_tumor_file@clinical.data$specimenID %in% specimens()])
+  
+    validate(need(length(tumor_sample_bc)>0, "No variant data found. Please modify your cohort."))
```

Comment:
With one gene i got this error: "Oncoplot requires at-least two genes for plottng." - can you add a `validate` statement to catch this error with a 'call to action' for the user?

Reply:
Ah! This is specific for Oncoplot because it plots multiple genes at once. Will make that explicit.